pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
 
+	var experimentalClient *client.ExperimentalClient
+	cfg, experimentalClientErr := clientConfig.ClientConfig()
```

Comment:
Maybe I'm not seeing it, but how does Factory creation fail? There is no if err != nil { return err } statement here. We are keeping the error and returning it when an experimental client is requested, not when the factory is created.

If client config panic's then that changes things, but it really should not panic.